host/xtest/regression_1000.c

```diff
@@ -1381,24 +1386,51 @@ static void xtest_tee_test_1017(ADBG_Case_t *c)
 	TEEC_ReleaseSharedMemory(&shm);
 }
 
+static void invoke_1byte_out_of_bounds(ADBG_Case_t *c, TEEC_Session *session,
+				      TEEC_SharedMemory *shm)
+{
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_PARTIAL_INPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	op.params[0].memref.parent = shm;
+	op.params[0].memref.size = shm->size / 2;
+	op.params[0].memref.offset = shm->size - (shm->size / 2) + 1;
+
+	/*
+	 * Since reference are not valid the invocation request should NOT
+	 * reach the TA whatever the status. Yet we expect noboby gets hurt.
+	 */
+	ADBG_EXPECT_NOT(c, TEEC_ERROR_TARGET_DEAD,
```

Comment:
I think we should expect that the driver catches this, so expect whatever error code we should get from that one.

Reply:
Expecting `TEEC_ERROR_BAD_PARAMETERS` or `TEEC_ERROR_GENERIC` should be good. libteec is currently reporting all errors by `ENOMEM` as `TEEC_ERROR_GENERIC`. If we ever update `ioctl_errno_to_res()` with better granularity I expect that `EINVAL` would be translated to `TEEC_ERROR_BAD_PARAMETERS`.